rollup/types/encoding/codecv1/codecv1.go

```diff
@@ -221,12 +218,7 @@ func (c *DAChunk) Hash() (common.Hash, error) {
 }
 
 // NewDABatch creates a DABatch from the provided encoding.Batch.
-func NewDABatch(batch *encoding.Batch) (*DABatch, error) {
-	// this encoding can only support a fixed number of chunks per batch
-	if len(batch.Chunks) > MaxNumChunks {
-		return nil, fmt.Errorf("too many chunks in batch")
-	}
```

Comment:
This was not removed in the monorepo yet

Reply:
I changed this check to another place: https://github.com/scroll-tech/go-ethereum/blob/64786f36fb5ca745a5e2219f0661da6d98776b59/rollup/types/encoding/codecv1/codecv1.go#L290-L292

the motivation is to reduce calling `getMaxNumChunks`, it's OK to move this check back here.